payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -86,27 +89,17 @@ internal abstract class BaseAddPaymentMethodFragment(
 
         childFragmentManager.addFragmentOnAttachListener { _, fragment ->
             (fragment as? ComposeFormDataCollectionFragment)?.let { formFragment ->
-                formFragment.paramMapLiveData.observe(viewLifecycleOwner) { paramMap ->
-                    sheetViewModel.updateSelection(
-                        paramMap?.run {
-                            PaymentMethodCreateParams.Type.fromCode(paramMap["type"] as String)
-                                ?.let {
-                                    PaymentMethodCreateParams(
-                                        it,
-                                        overrideParamMap = paramMap,
-                                        productUsage = setOf("PaymentSheet")
-                                    )
-                                }
-                        }?.let {
-                            PaymentSelection.New.GenericPaymentMethod(
-                                selectedPaymentMethod.displayNameResource,
-                                selectedPaymentMethod.iconResource,
-                                it,
-                                false
-                            )
-                        }
+                formFragment.formViewModel.completeFormValues.map { formFieldValues ->
```

Comment:
This logic was split into two functions one below in this class transformToPaymetMethodCreateParams and the other part to TransformToPaymentMethodCreateParams.